### ISSUE TITLE
Multiline comment support

### ISSUE THREAD

**Issue by msk4862** [otakustay/react-diff-view#180]
Is there a support of multiline comments just like Github has where you can select multiple lines of code and add comment for those selected lines. 
If its not there then can some someone please give some idea on how should I go about it?

**Comment by otakustay** [comment]
At the time this component is developed, GitHub didn't offer a multi-line commnet, however this can be achieved with the multi-line selection and custom widget, would you want a demo for that?

**Comment by msk4862** [comment]
Thanks @otakustay , but demo won't be required.
I'm thinking to achieve it with `gutterEvents` something like this, 
 
 ```
<Hunk
  key={`+${hunk.content}`}
  hunk={hunk}
  data-automation="hunk"
  gutterEvents={{
    onMouseDown({ change }) {
        const { isNormal, lineNumber, newLineNumber } = change;
        gutterLine.current = isNormal ? newLineNumber : lineNumber;
      },
      onMouseUp({ change }) {
        const { isNormal, lineNumber, newLineNumber } = change;
        const finalLine = isNormal ? newLineNumber : lineNumber;

        if (!gutterLine.current || gutterLine.current === finalLine) {
          // clicked on the same line
          console.log('clicked on the same line', gutterLine.current, finalLine);
        } else {
          // selected multiple lines

          console.log(
            'selected multiple line',
            `Initial: ${gutterLine.current} Final: ${finalLine}`
          );
        }

        gutterLine.current = null;
      },
  }}
  widgets={widgets(newPath)}
  />
```

But now I'm thinking on how should I highlight these selected lines, because in Github they have one custom property to have gutter line number and one can use these property to access any particular gutter and add any class name on it for any any custom css, but I don't see such property here but only a text of line number.

**Github**
<img width="1745" alt="Screenshot 2022-10-31 at 3 00 52 PM" src="https://user-images.githubusercontent.com/24875366/198977093-a8e3f85d-b16e-4ca5-80e7-4d0b6ddbf624.png">

**react-diff-view**
<img width="1279" alt="Screenshot 2022-10-31 at 3 10 17 PM" src="https://user-images.githubusercontent.com/24875366/198978383-f16c9a54-4782-430a-8d1a-266de66e304a.png">

**Comment by msk4862** [comment]
hey @otakustay , did you got time to build this demo? I think It will really help.

**Comment by otakustay** [comment]
I've been stuck on the multi-selection issue for a long time, since diff is not a sequencial lines, it is an interleaved structure with old-new changes side by side, it's quite difficult to determine whether a bunch of selected changes combines to a "range that can be defined with `{startLineNumber, endLineNumber}`", this blocks further investigation of comment widgets.

Hopefully, this is a pure data algorithm issue, not releated to UI representations, I think I can solve it recently.

**Comment by otakustay** [comment]
A little progress to this issue, I've published `2.5.0` to expose DOM event object in `gutterEvents` and `codeEvents`.

By this improvement, I wrote a deadly simple hook in our demo site to allow select a range of changes by holding SHIFT key and click on lines, you can see the [implement and usage here](https://github.com/otakustay/react-diff-view/commit/2e3fa74067b9c0753d1f59ed599b6b332b0d69fd#diff-eddfe7ac01f1ca79584c2d5dce5a3d3f78639f0a6c26074d7ffa33497c0d712a) and have a try on [our demo site](https://otakustay.github.io/react-diff-view/)

https://user-images.githubusercontent.com/639549/202982299-492ce9c7-4681-4cc0-b8be-404ed9a1ea91.mp4

I need to implement a more powerful comment widget to allow comment on range selected changes, this is not that difficult since `selectedChanges` is already generated by `useSelection`, just comparing the commented change and the last item in `selectedChanges`, I'll try to write it.

**Comment by msk4862** [comment]
hey @otakustay, First of all Thanks for going ahead and making the changes also, really appreciate it. 
and sorry, for being inactive for a while I got busy in some other things.  

I checked your implementation and selecting lines with shift key does look good but I wanted to know If we can have something more similar to what Github has which is selecting lines by clicking and dragging the mouse? 
Let me know your thoughts on this and I'm also going through the codebase to check the possibilities.
